ToolBox/PsychSignalDataConverter/PsychSignalDataConverter.cs

```diff
@@ -0,0 +1,270 @@
+        /// Should be called once the data has finished processing
+        /// </summary>
+        public void FlushAll()
+        {
+            foreach (var handle in _fileHandles.Values)
+            {
+                handle.Dispose();
+            }
+        }
+        
+        /// <summary>
+        /// Utility method to compresses the data contained with the psychsignal alternative data folder 
+        /// to a structure similar to equity minute files (e.g. /[symbol]/20010101.zip#20010101.csv)
+        /// </summary>
+        private void CompressData()
+        {
+            var finalPath = Path.Combine(Globals.DataFolder, "alternative", "psychsignal");
+            foreach (var tickerFolder in Directory.GetDirectories(finalPath))
+            {
+                foreach (var dataFile in Directory.GetFiles(tickerFolder, "*.csv", SearchOption.TopDirectoryOnly))
+                {
+                    Compression.Zip(dataFile, deleteOriginal: true);
+                    Log.Trace($"PsychSignalDataConverter.CompressData(): Successfully compressed: {dataFile}");
+                }
+            }
+        }
+        
+        /// <summary>
+        /// Handle to a file so that we don't have to open and close it every time we want
+        /// to write to a file. This helps us speed up time spent processing massively.
+        /// </summary>
+        private class TickerData : IDisposable
+        {
+            private readonly DirectoryInfo _destinationDirectory;
+            private readonly string _ticker;
+
+            private StreamWriter _writer;
+            private string _tempPath;
+            private DateTime _date;
+
+            /// <summary>
+            /// Creates writer instances and saves file path.
+            /// Used to keep file open until the path changes for the symbol
+            /// </summary>
+            /// <param name="ticker"></param>
+            /// <param name="date"></param>
+            /// <param name="destinationDirectory"></param>
+            public TickerData(string ticker, DateTime date, DirectoryInfo destinationDirectory)
+            {
+                _date = date;
+                _ticker = ticker;
+                // Using win32 Path.GetTempFileName can cause filename collisions
+                _tempPath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+                _writer = new StreamWriter(_tempPath);
+                _destinationDirectory = destinationDirectory;
+            }
+            
+            /// <summary>
+            /// Adds a new line to the writer
+            /// </summary>
+            /// <param name="timestamp">Event timestamp in UTC</param>
+            /// <param name="csv">CSV enumerable</param>
+            public void Append(DateTime timestamp, IEnumerable<string> csv)
+            {
+                var date = timestamp.Date;
+
+                if (date != _date)
+                {
+                    Dispose();
```

Comment:
Disposal is intended to be permanent, non-reversible action taken against on object. Perhaps `Dispose` here is really a `Recycle`. In general, once `Dispose` is called on an object, any further calls to the object have undefined behavior, but typically they will raise an `ObjectDisposedException`.